File: server/src/auth/dto/register.dto.ts

import { IsEmail, IsNotEmpty, IsString, MinLength } from 'class-validator';

export class RegisterDTO {
  @IsEmail({}, { message: 'Please enter a valid email address' })
  @IsNotEmpty({ message: 'Email should not be empty' })
  email: string;

  @IsString({ message: 'Please enter a valid first name' })
  @IsNotEmpty({ message: 'First name should not be empty' })
  firstName: string;

  @IsString({ message: 'Please enter a valid last name' })
  @IsNotEmpty({ message: 'Last name should not be empty' })
  lastName: string;

  @IsString({ message: 'Please enter a valid password' })
  @MinLength(6, { message: 'Password should be at least 6 characters long' })
  password: string;
}

File: client/lib/models/dto/auth/register.dto.dart
import 'package:client/models/dto/dto.dart';

class RegisterDTO extends DTO {
  String email;
  String firstName;
  String lastName;
  String password;

  RegisterDTO({
    required this.email,
    required this.firstName,
    required this.lastName,
    required this.password,
  });

  @override
  Map<String, dynamic> toMap() {
    return {
      'email': email,
      'firstName': firstName,
      'lastName': lastName,
      'password': password,
    };
  }
}
